src/libsyncengine/update_detection/update_detector/node.h

```diff
@@ -29,18 +30,39 @@
 namespace KDC {
 
 class Node {
+    public:
+        class MoveOriginInfos {
+            public:
+                MoveOriginInfos() = default;
+                MoveOriginInfos(const MoveOriginInfos &) = default;
+                MoveOriginInfos(const SyncPath &path, const NodeId &parentNodeId) :
+                    _path(path), _parentNodeId(parentNodeId), _isValid(true) {}
+
+                bool isValid() const;
+                const SyncPath &path() const;
+                const NodeId &parentNodeId() const;
+
+            private:
+                bool _isValid = false;
+                SyncPath _path = ":\0/:\0"; // Invalid path for increased safety
```

Comment:
Is there a reason for this specific pattern?

Reply:
This is an invalid path on Windows, macOS, and Linux (based on `platformInconsistencyCheckerWorker` -> `forbiddenFilenameChars`).

This should work as well:
```suggestion
                SyncPath _path = ":\0///"; // Invalid path for increased safety
```

It is quite the last safety net. This path should never be getted (if that happens, we will have a sentry, along with assertions and logs).